server/container/guice/guice-common/src/main/java/org/apache/james/StartUpChecksPerformer.java

```diff
@@ -0,0 +1,187 @@
+                    RequireResultType checkName(CheckName name);
+                }
+
+                @FunctionalInterface
+                interface RequireResultType {
+                    ReadyToBuild resultType(ResultType resultType);
+                }
+
+                static class ReadyToBuild {
+                    private final CheckName name;
+                    private final ResultType resultType;
+                    private Optional<String> description;
+
+                    ReadyToBuild(CheckName name, ResultType resultType) {
+                        this.name = name;
+                        this.resultType = resultType;
+                        this.description = Optional.empty();
+                    }
+
+                    public ReadyToBuild description(String description) {
+                        this.description = Optional.ofNullable(description);
+                        return this;
+                    }
+
+                    public CheckResult build() {
+                        return new CheckResult(name, resultType, description);
+                    }
+                }
+
+            }
+
+            public static Builder.RequireCheckName builder() {
+                return name -> resultType -> new Builder.ReadyToBuild(name, resultType);
+            }
+
+            private final CheckName name;
+            private final ResultType resultType;
+            private final Optional<String> description;
+
+            private CheckResult(CheckName name, ResultType resultType, Optional<String> description) {
+                Preconditions.checkNotNull(name);
+                Preconditions.checkNotNull(resultType);
+                Preconditions.checkNotNull(description);
+
+                this.name = name;
+                this.resultType = resultType;
+                this.description = description;
+            }
+
+            public CheckName getName() {
+                return name;
+            }
+
+            public ResultType getResultType() {
+                return resultType;
+            }
+
+            public Optional<String> getDescription() {
+                return description;
+            }
+
+            public boolean isBad() {
+                return resultType.equals(ResultType.BAD);
+            }
+
+            @Override
+            public String toString() {
+                return MoreObjects.toStringHelper(this)
+                    .add("name", name)
+                    .add("resultType", resultType)
+                    .add("description", description)
+                    .toString();
+            }
+        }
+
+        CheckResult check();
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StartUpChecksPerformer.class);
+
+    private final Set<StartUpCheck> startUpChecks;
+
+    @Inject
+    public StartUpChecksPerformer(Set<StartUpCheck> startUpChecks) {
+        this.startUpChecks = startUpChecks;
+    }
+
+    public void performCheck() {
```

Comment:
no test for that ?
What about having a StartUpChecks class that encapsulate that logic?